crates/bridge-exec/src/handler.rs

```diff
@@ -122,9 +122,13 @@ where
         let raw_message = borsh::to_vec::<BridgeMessage>(&signed_message)
             .expect("should be able to borsh serialize raw message");
 
-        self.l2_rpc_client
-            .submit_bridge_msg(raw_message.into())
-            .await?;
+        let l2_rpc_client = self
+            .l2_rpc_client_pool
+            .get()
+            .await
+            .map_err(|err| ExecError::WsPool(err))?;
```

Comment:
Maybe make an accessor on the type that wraps this repeated pattern so you can just do something like `self.get_ready_rpc_client().await?`.